internal/s3storage/s3storage.go

```diff
@@ -68,3 +163,65 @@ func (s *Service) CreateArtifact(artifactSpec artifact.Spec, md5 string) (string
 
 	return uploadURL, nil
 }
+
+func (s *Service) Close() error {
+	if s.sqsHandlerQuitChannel != nil {
+		close(s.sqsHandlerQuitChannel)
+	}
+	if s.sqsHandlerErrorChannel != nil {
+		err := <-s.sqsHandlerErrorChannel
+		return err
+	}
+	return nil
+}
+
+func (s *Service) startSQSHandler(handler func(msg string) error) {
+	log.Infof("starting SQS handler")
+	s.sqsHandlerQuitChannel = make(chan struct{})
+	s.sqsHandlerErrorChannel = make(chan error, 1)
+	go func() {
+		for {
+			select {
+			case _, ok := <-s.sqsHandlerQuitChannel:
+				if !ok {
+					s.sqsHandlerErrorChannel <- nil
+					return
+				}
+			default:
+				// continue
+			}
```

Comment:
Could this not just be an `if` statement?

```suggestion
			_, ok := <-s.sqsHandlerQuitChannel:
			if !ok {
				s.sqsHandlerErrorChannel <- nil
				return
			}
```

Reply:
I found that the `select` does not block, but `_, ok := <-s.sqsHandlerQuitChannel:` does block. Thats why it looks like it does.

Reply:
Yeah, good point. My mistake 👍